### PR TITLE
miner, cmd, eth: require explicit etherbase address

### PR DESCRIPTION
**Pull request by rjl493456442**
This PR introduces a breaking change to `miner.etherbase` flag.

Previously users can configure etherbase address by explicitly setting the flag,
or the first local account will be used as etherbase automatically. In this PR, 
this "default first account" feature is removed, namely users must explicitly
set the etherbase by command line flag.

In Proof-of-stake world, the fee recipient address is provided by CL, namely
it's not configured in Geth side anymore. So `miner.etherbase` is mostly for
legacy networks(pow, clique networks etc).

### CODE REVIEW

**Review by holiman — APPROVED** [review]
LGTM

**Review by holiman — CHANGES_REQUESTED** [review]

File: miner/miner_test.go [hunk 1]
```diff
@@ -207,21 +210,16 @@ func TestMinerSetEtherbase(t *testing.T) {
...
-	miner.Start(common.HexToAddress("0x1337"))
+	miner.Start()
 	// Stop the downloader and wait for the update loop to run
 	mux.Post(downloader.DoneEvent{})
-
 	waitForMiningState(t, miner, true)
-	// The miner should now be using the good address
-	if got, exp := miner.coinbase, common.HexToAddress("0x1337"); got != exp {
-		t.Fatalf("Wrong coinbase, got %x expected %x", got, exp)
-	}
```
[the remark this answers is not in the document]
Comment (holiman) [reply miner/miner_test.go:224]: > If etherbase is updated later via console, the new etherbase address should be used

Yes, please add a check for that

[the remark this answers is not in the document]
Comment (rjl493456442) [reply miner/miner_test.go]: added

**Review by holiman — APPROVED** [review]
LGTM

File: miner/worker.go [hunk 2]
```diff
@@ -1114,11 +1123,11 @@ func (w *worker) commitWork(interrupt *int32, noempty bool, timestamp int64) {
 	// Set the coinbase if the worker is running or it's required
 	var coinbase common.Address
 	if w.isRunning() {
-		if w.coinbase == (common.Address{}) {
+		coinbase = w.etherbase()
+		if coinbase == (common.Address{}) {
```
Comment (holiman) [inline miner/worker.go:1127]: But - this still requires an etherbase, right? If the work-package from CL contains an etherbase, this shouldn't be needed, should it?

Comment (rjl493456442) [reply miner/worker.go:1127]: `commitWork` is only for legacy block generation.
`generateWork` is for post-merge mode.

**Pull request merged**